File: pinyougou-sellergoods-interface/src/main/java/com/lsf/pinyougou/sellergoods/service/GoodsDescService.java
package com.lsf.pinyougou.sellergoods.service;

import java.util.List;

import com.lsf.pinyougou.pojo.TbGoodsDesc;
import vo.PageResult;


/**
 * 服务层接口
 */
public interface GoodsDescService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbGoodsDesc> findAll();


    /**
     * 无条件分查询
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 多条件分页查询
     *
     * @param pageNum  当前页
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPageLimit(TbGoodsDesc goodsDesc, int pageNum, int pageSize);


    /**
     * 添加
     */
    void add(TbGoodsDesc goodsDesc);


    /**
     * 修改
     */
    void update(TbGoodsDesc goodsDesc);


    /**
     * 根据 ID 获取实体
     *
     * @param id
     * @return
     */
    TbGoodsDesc findOne(long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void batchDelete(Long[] ids);

}

File: pinyougou-manager-web/src/main/webapp/js/service/goodsDescService.js
// 服务层
pyg.service('goodsDescService', function ($http) {

    // 查询所有数据
    this.findAll = function () {
        return $http.get('../goodsDesc/findAll.do');
    };

    // 无条件分页查询
    this.findPage = function (page, size) {
        return $http.get('../goodsDesc/findPage.do?page=' + page + '&size=' + size);
    };

    // 根据 ID 查询实体
    this.findOne = function (id) {
        return $http.get('../goodsDesc/findOne.do?id=' + id);
    };

    // 增加
    this.add = function (entity) {
        return $http.post('../goodsDesc/add.do', entity);
    };

    // 修改
    this.update = function (entity) {
        return $http.post('../goodsDesc/update.do', entity);
    };

    // 批量删除
    this.batchDelete = function (selectedList) {
        return $http.get('../goodsDesc/batcheDelete.do?ids=' + selectedList);
    };

    // 多条件分页查询
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('../goodsDesc/search.do?page=' + page + "&size=" + size, searchEntity);
    };
});
